design/workitems.go

```diff
@@ -150,4 +150,19 @@ var _ = a.Resource("workitem", func() {
 		a.Response(d.NotFound, JSONAPIErrors)
 		a.Response(d.Unauthorized, JSONAPIErrors)
 	})
+	a.Action("reorder", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.PATCH("/reorder"),
+		)
+		a.Description("reorder the work item with the given id.")
+		a.Payload(workItemSingle)
```

Comment:
@aslakknutsen I'm going over your concerns to see if this PR can be merged. Meanwhile the API has changed to return an `a.ArrayOf(workItem2)` which is fine.